### ISSUE TITLE
Prevent capitalization of email address inputs

### ISSUE THREAD

**Issue by zarino** [mysociety/writetothem#256]
Very annoying when I type an email (twice!!!) and my phone/tablet auto-capitalizes the first letter, meaning I have to go back and uncapitalize it.

![image](https://cloud.githubusercontent.com/assets/739624/8496927/d885a0b0-2170-11e5-8397-3f4d2b8890b0.jpg)

WriteToThem could tell my browser not to capitalize the email field by adding an attribute of `autocapitalize="off"`. Simples!

**Comment by zarino** [comment]
Context: 36% of visitors to WTT use a phone or tablet, and of them, 60% use an iPad or iPhone, which default to capitalizing user text input (it can be turned off in the device settings, but who does that?). No idea whether Android behaves similarly.

**Comment by dracos** [comment]
Would making it type=email fix this and also make the keyboard even better for entering an email address?

**Comment by zarino** [comment]
@dracos – yes, it looks like `type="email"` automatically implies no capitalization. So let's do that!

**Comment by zarino** [comment]
The offending line is:

* https://github.com/mysociety/writetothem/blob/master/web/write.php#L219

A quick grep of my work directory pulled out a few other mySociety sites that have an `<input type="text" name="email">`…

* [theyworkforyou www/docs/user/password/index.php](https://github.com/mysociety/theyworkforyou/blob/604bbb041c36cc5e84d83f2a214621da57cbe144/www/docs/user/password/index.php#L117)
* [theyworkforyou www/includes/easyparliament/page.php](https://github.com/mysociety/theyworkforyou/blob/604bbb041c36cc5e84d83f2a214621da57cbe144/www/includes/easyparliament/page.php#L1161)
* [theyworkforyou www/includes/easyparliament/templates/html/alert/postcode.php](https://github.com/mysociety/theyworkforyou/blob/094bc561723871ff4bdd18e69d7029959a54a906/www/includes/easyparliament/templates/html/alert/postcode.php#L70-L72)
* [theyworkforyou www/includes/easyparliament/templates/html/alert/update-mp.php](https://github.com/mysociety/theyworkforyou/blob/d688e18dc2706e37eb584e84b60fdda4cdb72f87/www/includes/easyparliament/templates/html/alert/update-mp.php#L60-L62)
* [sayit.mysociety.org sayit_mysociety_org/templates/account/signup.html](https://github.com/mysociety/sayit.mysociety.org/blob/cbd9972893d64322f84ad88de7ffa98f8193e8c5/sayit_mysociety_org/templates/account/signup.html#L38)
* [poplus-home blog/index.html](https://github.com/poplus/home-poplus/blob/0a62c13a4c0219ebfab61b9478b611058d36675c/blog/index.html#L10)
* [fixmystreet templates/web/base/admin/body.html](https://github.com/mysociety/fixmystreet/blob/5d4c7be61ea01280fa618ca421dbee5b9e97f56a/templates/web/base/admin/body.html#L147)

I expect there may be others, like form inputs generated by Django template functions, or inputs with a `name` attribute other than `"email"`.

**Comment by zarino** [comment]
Looks like WriteToThem uses A PHP packages called [HTML_QuickForm](https://pear.php.net/package/HTML_QuickForm) to create the various inputs, and it doesn't support an `<input type="email">` element out of the box. We'll probably need to add a new one using [registerElementType()](http://pear.php.net/manual/en/package.html.html-quickform.html-quickform.registerelementtype.php).

**Comment by jacksonj04** [comment]
QuickForm is Gnarly Legacy Stuff™ (https://github.com/mysociety/commonlib/tree/master/phplib/HTML), and I remember swearing at it a lot when I implemented the redesign. I'm fairly sure I looked at updating it to QuickForm2, or at least updating it to its latest stable version, but didn't because it would have involved an awful lot of mess at the time.

It may well be that the codebase is now in a better state after a couple of rounds, and is more suited to ditching the old version and moving to a newer one altogether, rather than just applying a patch using `registerElementType()`. YMMV.
